sources/api/schnauzer/src/helpers.rs

```diff
@@ -422,6 +400,163 @@ pub fn default(
+    if helper.params().len() != 2 {
+        return Err(RenderError::from(
+            error::TemplateHelperError::IncorrectNumberOfParams {
+                expected: 4,
+                received: helper.params().len(),
+                helper: helper.name().to_string(),
+                template: template_name.to_owned(),
+            },
+        ));
+    }
+
+    // get the first parameter, which should be either 'account' or 'region'.
+    // convert this to a boolean is_account. error if param is bad.
+    let account_or_region_param = get_param(helper, 0)?;
+    let account_or_region =
+        account_or_region_param
+            .as_str()
+            .with_context(|| error::InvalidTemplateValue {
+                expected: "string",
+                value: account_or_region_param.to_owned(),
+                template: template_name.to_owned(),
+            })?;
+    let is_account = match account_or_region {
+        "account" => true,
+        "region" => false,
+        _ => {
+            return Err(RenderError::from(error::TemplateHelperError::EcrParam {
+                value: account_or_region.to_owned(),
+            }))
+        }
+    };
+
+    // get the second parameter, which is probably given by the template value
+    // settings.aws.region. regardless, we expect it to be a string.
+    let aws_region = get_param(helper, 1)?;
+    let aws_region = aws_region.as_str().ok_or_else(|| {
+        RenderError::from(error::TemplateHelperError::EcrRegion {
+            value: aws_region.to_owned(),
+            template: template_name.to_owned(),
+        })
+    })?;
+
+    // get a mapping of regions to their ECR accounts. find the pair for our
+    // region, or the fallback account and region if our region is not found.
+    let ecr_map = ecr_map();
+    let ecr_region_and_account = match ecr_map.get(aws_region) {
+        None => (ECR_FALLBACK_REGION, ECR_FALLBACK_ACCOUNT),
+        Some(account) => (aws_region, *account),
+    };
+
+    // write either the resultant account or resultant region to the template.
+    if is_account {
+        out.write(ecr_region_and_account.1)
+            .context(error::TemplateWrite {
+                template: template_name.to_owned(),
+            })?;
+    } else {
+        out.write(ecr_region_and_account.0)
+            .context(error::TemplateWrite {
+                template: template_name.to_owned(),
+            })?;
+    }
+    Ok(())
+}
+
+/// If there is no region setting at all, then we will error. But if there is a
+/// region setting that does not exist in our map (for example a new region is
+/// created or being tested), then we will fallback to pulling ECR containers
+/// from us-east-1.
+const ECR_FALLBACK_REGION: &str = "us-east-1";
+const ECR_FALLBACK_ACCOUNT: &str = "328549459982";
+
+/// A map to tell us which account to pull ECR images from for a given region.
+fn ecr_map() -> &'static HashMap<&'static str, &'static str> {
```

Comment:
```suggestion
```
Can we just do the `lazy_static!` at the top level rather than in a function?  It still won't be constructed until needed, and keeping it at the top level is more common.  Their example does a hashmap like this: https://docs.rs/lazy_static/1.4.0/lazy_static/#example